.github/workflows/check-sorbet-typing-mode.yml

```
@@ -0,0 +1,67 @@
+name: Sorbet
+on: # yamllint disable-line rule:truthy
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [main]  # Adjust this to the branch you want for push events
+
+jobs:
+  check-sorbet-typing-mode:
+    name: Sorbet Typing Mode Check
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: ${{ github.event_name == 'pull_request' && '2' || '0' }}
+
+      - name: Get Changed Files
+        id: changed-files
+        uses: tj-actions/changed-files@ed68ef82c095e0d48ec87eccea555d944a631a4c # v46
```

comment should include the full version pinned by the sha:
```suggestion
        uses: tj-actions/changed-files@ed68ef82c095e0d48ec87eccea555d944a631a4c # v46.0.5
```